addons/website/static/src/components/survey/survey.js

```diff
@@ -0,0 +1,514 @@
+                const paletteColor2 = state.allPalettes[paletteName]['color2'];
+                const delta1 = deltaE(color1, paletteColor1);
+                const delta2 = deltaE(color2, paletteColor2);
+                state.allPalettes[paletteName]['score'] = (delta1 + delta2) / 2;
+            }
+            palettes = Object.values(state.allPalettes).sort((a, b) => a['score'] - b['score']);
+            state.recommendedPalette = palettes[0];
+        } else {
+            palettes = Object.values(state.allPalettes);
+        }
+        const selectedPalettes = {};
+        palettes.slice(1, MAX_PALETTES + 1).forEach((palette) => {
+            selectedPalettes[palette.id] = palette;
+        });
+        state.palettes = selectedPalettes;
+    },
+    updateRecommendedThemes({state}, themes) {
+        state.themes = themes.slice(0, 3).map(({name, id, svg}) => {
+            return {name, id, svg};
+        });
+    }
+};
+
+const getters = {
+    getWebsiteTypes() {
+        return Object.values(WEBSITE_TYPES);
+    },
+
+    getSelectedType(_, id) {
+        return id ? WEBSITE_TYPES[id] : undefined;
+    },
+
+    getWebsitePurpose() {
+        return Object.values(WEBSITE_PURPOSES);
+    },
+
+    getSelectedPurpose(_, id) {
+        return id ? WEBSITE_PURPOSES[id] : undefined;
+    },
+
+    getFeatures({state}) {
+        return Object.values(state.features);
+    },
+
+    getPalettes({state}) {
+        return Object.values(state.palettes);
+    },
+};
+
+async function getInitialState(wid) {
+
+    const features = rpc.query({
+        model: 'website.survey.feature',
+        method: 'search_read',
+        fields: ['title', 'description', 'type', 'icon', 'website_types_preselection'],
+    }).then(function (results) {
+        const features = {};
+        results.forEach(result => {
+            features[result.id] = Object.assign({}, result, {selected: false});
+            const wtp = features[result.id].website_types_preselection;
+            features[result.id].website_types_preselection = wtp ? wtp.split(',') : [];
+        });
+        return features;
+    });
+
+    const logo = rpc.query({
+        model: 'website',
+        method: 'get_survey_logo',
+        args: [[parseInt(wid)]],
+    });
+    const industries = rpc.query({
+        model: 'website',
+        method: 'get_survey_industries',
+    }).catch((_) => false);
+
+    const allPalettes = {}, palettes = {};
+    const style = window.getComputedStyle(document.documentElement);
+    const themeNames = weUtils.getCSSVariableValue('theme-names', style).split(' ');
+    themeNames.forEach((rawName) => {
+        const name = rawName.replace(/'/g, '');
+        const count = weUtils.getCSSVariableValue(`o-count-${name}`, style);
+        for (let i = 1; i <= count; i++) {
+            const paletteName = `${name}-${i}`;
+            const palette = {};
+            palette['id'] = paletteName;
```

Comment:
dot access for fixed properties => `palette.id`, though in this case you can simply put that field in the palette declaration directly:
```js
const palette = {
  id: paletteName
}
```